client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/plan-only.tsx

```diff
@@ -23,13 +38,108 @@ const isMonthsOld = ( months: number, rawDate?: string ) => {
 	return moment().diff( parsedDate, 'months' ) > months;
 };
 
-export default function PlanOnlyThankYou( { primaryPurchase }: PlanOnlyThankYouProps ) {
+const PlanOnlyThankYou = ( {
+	primaryPurchase,
+	isEmailVerified,
+	errorNotice,
+	removeNotice,
+	successNotice,
+}: PlanOnlyThankYouProps ) => {
 	const siteId = useSelector( getSelectedSiteId );
 	const siteSlug = useSelector( getSelectedSiteSlug );
 	const siteCreatedTimeStamp = useSelector(
 		( state ) => getSiteOptions( state, siteId ?? 0 )?.created_at
 	);
 
+	const [ resendStatus, setResendStatus ] = useState( RESEND_NOT_SENT );
+
+	const verifyEmailNoticeId = 'ecommerce-verify-email-notice';
+	const resendEmail = () => {
+		removeNotice( verifyEmailNoticeId );
+
+		if ( RESEND_PENDING === resendStatus ) {
+			return;
+		}
+
+		setResendStatus( RESEND_PENDING );
+
+		wpcom.req.post( '/me/send-verification-email', ( error: Error ) => {
+			if ( error ) {
+				errorNotice( translate( "Couldn't resend verification email. Please try again." ), {
+					id: verifyEmailNoticeId,
+					duration: 5000,
+				} );
+
+				setResendStatus( RESEND_ERROR );
+				return;
+			}
+
+			successNotice( translate( 'Email sent' ), { id: verifyEmailNoticeId, duration: 5000 } );
+
+			setResendStatus( RESEND_SUCCESS );
+		} );
+	};
+
+	const resendButtonText = () => {
+		if ( resendStatus === RESEND_PENDING ) {
+			return translate( 'Sending…' );
+		}
+
+		return translate( 'Resend email' );
+	};
+
+	// At this point in the flow, having purchased a plan for a specific site,
+	// we can be confident that `siteId` is a number and not `null`
+	const siteAdminUrl = useSelector( ( state ) => getSiteWooCommerceUrl( state, siteId as number ) );
+	const emailAddress = useSelector( getCurrentUserEmail );
+
+	let subtitle;
+	let headerButtons;
+
+	if ( primaryPurchase.productSlug === 'ecommerce-bundle' ) {
```

Comment:
```suggestion
	if ( isWpComEcommercePlan( primaryPurchase.productSlug ) ) {
```
I missed this in the last round of review, we should be using the helper function to check for eCommerce plans rather than a hardcoded string. That ensures all types of eCommerce plans are caught regardless of payment interval. The `isWpComEcommercePlan` function is part of the `@automattic/calypso-products` package.

Reply:
Oh, cool. Thanks for this... definitely helpful for me to see these as I familiarize myself with the different types of helper functions we have in place!